pkg/apis/projectcalico/v3/nodestatus.go

```diff
@@ -0,0 +1,235 @@
+	V6Peers []CalicoNodePeer `json:"v6Peers,omitempty"`
+}
+
+// CalicoNodeBGPRouteStatus defines the observed state of routes status on the node.
+type CalicoNodeBGPRouteStatus struct {
+	// V4 represents IPv4 routes on the node.
+	V4Routes []CalicoNodeRoute `json:"v4Routes,omitempty"`
+
+	// V6 represents IPv6 routes on the node.
+	V6Routes []CalicoNodeRoute `json:"v6Routes,omitempty"`
+}
+
+// CalicoNodeBirdStatus defines the observed state of bird.
+type CalicoNodeBirdStatus struct {
+	// Ready indicates if bird status is ready.
+	Ready bool `json:"ready,omitempty"`
+
+	// Bird version.
+	Version string `json:"version,omitempty"`
+
+	// Router ID used by bird.
+	RouterID string `json:"routerID,omitempty"`
+
+	// ServerTime holds the value of serverTime from bird.ctl output.
+	ServerTime string `json:"serverTime,omitempty"`
+
+	// LastBootTime holds the value of lastBootTime from bird.ctl output.
+	LastBootTime string `json:"lastBootTime,omitempty"`
+
+	// LastReconfigTime holds the value of lastReconfigTime from bird.ctl output.
+	LastReconfigTime string `json:"lastReconfigTime,omitempty"`
+}
+
+// CalicoNodePeer contains the status of BGP peers on the node.
+type CalicoNodePeer struct {
+	// IP address of the peer whose condition we are reporting.
+	PeerIP string `json:"peerIP,omitempty"`
+
+	// Type indicates whether this peer is configured via the node-to-node mesh,
+	// or via en explicit global or per-node BGPPeer object.
+	Type BGPPeerType `json:"type,omitempty"`
+
+	// The state is the bird bgp session state.
+	State string `json:"state,omitempty"`
+
+	// Since the state or reason last changed.
+	Since string `json:"since,omitempty"`
+
+	// Extra information from bird on the bgp session.
+	Info string `json:"reason,omitempty"`
+}
+
+// CalicoNodeRoute contains the status of BGP routes on the node.
+type CalicoNodeRoute struct {
+	// Type indicates if the route is being used for forwarding or not.
+	Type CalicoNodeRouteType `json:"type,omitempty"`
+
+	// Destination of the route.
+	Destination string `json:"destination,omitempty"`
+
+	// Gateway for the destination.
+	Gateway string `json:"gateway,omitempty"`
+
+	// Interface for the destination
+	Interface string `json:"interface,omitempty"`
+
+	// LearnedFrom indicates who installed this route.
+	// If it is populated by a BGP peer, this is the name of the BGPPeer object.
+	// If it is populated by node mesh, this is the name of the node.
+	// Or it is one of kernel, direct or static.
+	LearnedFrom CalicoNodeRouteLearnedFrom `json:"learnedFrom,omitempty"`
+}
+
+// CalicoNodeRouteLearnedFrom contains the information of the source from which a routes has been learned.
+type CalicoNodeRouteLearnedFrom struct {
+	// Type of the source where a route is learned from.
+	SourceType CalicoNodeRouteSourceType `json:"sourceType,omitempty"`
+
+	// If sourceType is NodeMesh or BGPPeer, IP address of the router that sent us this route.
+	PeerIP string `json:"peerIP,omitempty"`
```

Comment:
Will this be just an IP address or an IP and port? The comment suggests this is just an IP, which is probably OK.

Reply:
It is just an IP.

Reply:
Added validation of `validate:"omitempty,ip"`